### PR TITLE
Update ResolveChannelFromHostFunc to take in a string instead of URL

### PR DESCRIPTION
**Pull request by shefaliv**
Basically applying https://github.com/knative/eventing/pull/2458 to this repo as well.

Also updating the sdk-go dependency to match the eventing repo.

### CODE REVIEW

File: vendor/knative.dev/eventing/pkg/channel/event_receiver.go [hunk 1]
```diff
@@ -63,7 +62,7 @@ type ReceiverOptions func(*EventReceiver) error
 
 // ResolveChannelFromHostFunc function enables EventReceiver to get the Channel Reference from incoming request HostHeader
 // before calling receiverFunc.
-type ResolveChannelFromHostFunc func(url.URL) (ChannelReference, error)
+type ResolveChannelFromHostFunc func(string) (ChannelReference, error)
```
Comment (grantr) [inline vendor/knative.dev/eventing/pkg/channel/event_receiver.go]: I think this change will need to be propagated to consuming code in this PR too

Comment (shefaliv) [reply vendor/knative.dev/eventing/pkg/channel/event_receiver.go]: Yeah looks like it =/.  I was trying to keep them separate initially. Updated the PR!

**Comment by grantr** [comment]
This is effectively a revert of #839

**Comment by grantr** [comment]
Build failure:
```
natss/cmd/channel_dispatcher/main.go:109:41: not enough arguments in call to tracing.SetupDynamicPublishing
	have (*zap.SugaredLogger, *configmap.InformedWatcher, string)
	want (*zap.SugaredLogger, *configmap.InformedWatcher, string, string)
```

This change is related to https://github.com/knative/eventing/pull/2333. SetupDynamicPublishing now takes an argument for the name of the tracing configmap. You can see the required change to callers in https://github.com/knative/eventing/pull/2333/files#diff-dd94eee2a6a728e7b3030b641a14835bR128.

https://github.com/knative/eventing-contrib/blob/595517fb3ef017f8c44fff6c77a2a7069e8c91f2/natss/cmd/channel_dispatcher/main.go#L109

needs to change to

```go
if err = tracing.SetupDynamicPublishing(logger, opt.ConfigMapWatcher, "natss-ch-dispatcher", tracingconfig.ConfigName); err != nil { 
```

and this import needs to be added

```go
tracingconfig "knative.dev/pkg/tracing/config"
```

File: natss/cmd/channel_dispatcher/main.go [hunk 2]
```diff
@@ -106,7 +107,7 @@ func main() {
 	opt.ConfigMapWatcher.Watch(metrics.ConfigMapName(), metrics.UpdateExporterFromConfigMap(component, logger))
 
 	// Setup zipkin tracing.
-	if err = tracing.SetupDynamicPublishing(logger, opt.ConfigMapWatcher, "natss-ch-dispatcher"); err != nil {
+	if err = tracing.SetupDynamicPublishing(logger, opt.ConfigMapWatcher, "natss-ch-dispatcher", tracingconfig.ConfigName); err != nil {
```
Comment (mattmoor) [inline natss/cmd/channel_dispatcher/main.go]: Remove trailing whitespace:
```suggestion
	if err = tracing.SetupDynamicPublishing(logger, opt.ConfigMapWatcher, "natss-ch-dispatcher", tracingconfig.ConfigName); err != nil {
```

**Pull request merged**